src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(2);
```

Comment:
When slicing the 'YYYY' to get 'YY', consider adding a check to ensure 'YYYY' is actually 4 characters long to avoid potential errors.